src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/TcpExpiredServerCertResource .cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Net;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.Security.Cryptography.X509Certificates;
+using WcfTestBridgeCommon;
+using WcfService.CertificateResources;
+using System.Collections.Generic;
+
+namespace WcfService.TestResources
+{
+    internal class TcpExpiredServerCertResource : TcpResource
+    {
+        protected override string Address { get { return "tcp-ExpiredServerCert"; } }
+
+        protected override Binding GetBinding()
+        {
+            NetTcpBinding binding = new NetTcpBinding() { PortSharingEnabled = false };
+            binding.Security.Mode = SecurityMode.Transport;
+            binding.Security.Transport.ClientCredentialType = TcpClientCredentialType.None;
+
+            return binding;
+        }
+
+        protected override void ModifyHost(ServiceHost serviceHost, ResourceRequestContext context)
+        {
+            // Ensure the https certificate is installed before this endpoint resource is used
+            //Create an expired certificate
+            var fqdn = Dns.GetHostEntry("127.0.0.1").HostName;
+            var hostname = fqdn.Split('.')[0];
+            CertificateCreationSettings certificateCreationSettings = new CertificateCreationSettings()
+            {
+                IsValidCert = false,
+                ValidityNotBefore = DateTime.UtcNow - TimeSpan.FromDays(4),
+                ValidityNotAfter = DateTime.UtcNow - TimeSpan.FromDays(2),
+                Subjects = new string[] { fqdn, hostname, "localhost" }
+            };
```

Comment:
Why is "localhost" here?  Does this work properly across machines (Windows to Windows)?